znvis/particle/particle.py

```
@@ -113,35 +117,97 @@ def construct_mesh_list(self):
-            self.mesh_list.append(mesh)
+                    raise ValueError(
+                        "Director shape does not match number of position frames."
+                    )
+
+        # Check data for consistency
+        for i, position in enumerate(self.position):
+            if np.isnan(position).any():
+                raise ValueError(
+                    f"The provided position data contains at least one "
+                    f"NaN value at time step {i}."
+                )
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)